napari/layers/_layer_actions.py

```diff
@@ -47,7 +48,10 @@ def _split_stack(ll: LayerList, axis: int = 0):
     if layer.rgb:
         images = stack_utils.split_rgb(layer)
     else:
-        images = stack_utils.stack_to_images(layer, axis)
+        order = current_viewer().dims.order
```

Comment:
while `current_viewer` has been a necessary evil in a few places, i'd like to think we could do better here (where we have access to the `LayerList`).  So far, there's been a 1:1 correspondence between viewer and LayerList, so in all but some extreme cases (like testing) `LayerList` could easily know about the viewer it is in (much like all widgets know their `parent()`).  

@sofroniewn, @jni, what would you think about adding a viewer reference to `LayerList`?  could be stored as a weakref and retrieved in a property getter?

Reply:
It's a good question. I think there is definitely the possibility of multiple viewers sharing a single layer list, e.g. orthoviewers.

One question is what happens here if there is no viewer instantiated?

Ultimately, it might be interesting here to think about what it would look like to provide the axis parameter from the context menu, rather than search for some global (whether that global is grabbed with `current_viewer` or a reference to a parent (that may or may not be "current"). E.g. with a sub-menu?

Reply:
> One question is what happens here if there is no viewer instantiated?

yeah, that's why I made it ` -> Optional[Viewer]` ... though rare, in this particular case, I think we's just fall back to what we had before.

> it might be interesting here to think about what it would look like to provide the axis parameter from the context menu, rather than search for some global (whether that global is grabbed with current_viewer or a reference to a parent (that may or may not be "current"). E.g. with a sub-menu?

I agree, this is probably the most pertinent to the issue at hand (not dissimilar from our `Project` menu, where we also need to provide the ability to pick an axis)

i do think the parent viewer concept would be potentially useful elsewhere though, but need to think about the multi-viewer case

Reply:
Thinking about this a bit more, I very much think it would be better to let the UI select a parameter. (a) the UI knows about the layer, so it doesn't have to do weird logic in case the layer doesn't have the right number of dims; and (b) in the multicanvas, orthoviews model, "current_viewer()" might need to be replaced by "current_canvas" and honestly it all just gets a bit messy. Better to just let the user provide the parameter, e.g. in a nested menu.

Reply:
@tlambert03 can the nested menu change during runtime so each n-dimensional layer has its own n axes for splitting?

Reply:
yes, it could... but I think it might be easier (in the sense of flexibility/simplicity) to use a modal dialog rather than a nested menu.  I know we've generally had an aversion to all things pop-up.  But in some recent conversations it's been agreed that "modal dialogs that come and go immediately" are ok, and this feels like a good case for that.   So, I'm kinda thinking of maybe two menu items:

```
Split Image >
    split along first axis
    split arbitrary axis ...
```

not sure about that, but if we always pop up a dialog, it could be annoying for the rather common case of splitting axis 0.

I'm not 100% sure that we want to avoid a lot of layer-specific menu customization, but it'd be nice if we could punt on that a bit